test/test_proxy_tensor.py

```diff
@@ -987,7 +985,15 @@ def f(a, b):
             assert b.shape[0] == 8
             return a.cos()
         fx_g = make_fx(f, tracing_mode="symbolic")(torch.randn(16), torch.randn(8))
-        self.assertExpectedInline(str(fx_g.shape_env.get_guard_expr()), "Eq(s1, 8) & Eq(s0, 2*s1)")
+        self.assertExpectedInline(str(fx_g.shape_env.get_guard_expr()).replace(" & ", "\n"), """\
+Eq(t0.size(0), s0)
+Eq(t0.storage_offset(), 0)
```

Comment:
Oh I guess this is a cost of always 0/1 specializing - even if our function happened to not use storage_offset/stride in a meaningful way, we still specialize and generate guards for it.

Curiosity question: does inductor generate a meaningfully different kernel for 0/1 strides/storage_offset (so specializing in cases where it seems unnecessary like in this function might still be profitable)?

Reply:
Zero offset would avoid an addition instruction in the pointer arithmetic. But the benefit is super marginal (since you can hoist this add out of the kernel entirely) so I don't recommend we do it